lib/rules/new-cap.js

```diff
@@ -60,6 +61,21 @@ function calculateCapIsNewExceptions(config) {
     return capIsNewExceptions.reduce(invert, {});
 }
 
+/**
+ * Gets the source code of a given node, excludes spaces and comments.
+ * @param {ASTNode} node - A node to get.
+ * @param {RuleContext} context - The current context.
+ * @returns {string} the source code of the node.
+ */
+function getSource(node, context) {
+    var source = "";
```

Comment:
This is not a good way to compare. There's no guarantee that there's isn't extra whitespace or comments in between tokens. You should just use `context.getSource`